server/monitor/src/main/resources/resources/functions.js

```diff
@@ -0,0 +1,686 @@
+      $(this).parent().removeClass('active');
+      sessionStorage.autoRefresh = 'false';
+    } else {
+      $(this).parent().addClass('active');
+      sessionStorage.autoRefresh = 'true';
+    }
+  });
+}
+
+/**
+ * Global timer that checks for auto refresh status every 5 seconds
+ */
+TIMER = setInterval(function() {
+  if (sessionStorage.autoRefresh == 'true') {
+    $('.auto-refresh').parent().addClass('active');
+    refresh();
+    refreshNavBar();
+  } else {
+    $('.auto-refresh').parent().removeClass('active');
+  }
+}, 5000);
+
+/**
+ * Empty function in case there is no refresh implementation
+ */
+function refresh() {
+}
+
+/**
+ * Converts a number to a size with suffix
+ *
+ * @param {number} size Number to convert
+ * @return {string} Number with suffix added
+ */
+function bigNumberForSize(size) {
+  if (size === null)
+    size = 0;
+  return bigNumber(size, SIZE_SUFFIX, 1024);
+}
+
+/**
+ * Converts a number to a quantity with suffix
+ *
+ * @param {number} quantity Number to convert
+ * @return {string} Number with suffix added
+ */
+function bigNumberForQuantity(quantity) {
+  if (quantity === null)
+    quantity = 0;
+  return bigNumber(quantity, QUANTITY_SUFFIX, 1000);
+}
+
+/**
+ * Adds the suffix to the number, converts the number to one close to the base
+ *
+ * @param {number} big Number to convert
+ * @param {array} suffixes Suffixes to use for convertion
+ * @param {number} base Base to use for convertion
+ * @return {string} The new value with the suffix
+ */
+function bigNumber(big, suffixes, base) {
+  // If the number is smaller than the base, return thee number with no suffix
+  if (big < base) {
+    return big + suffixes[0];
+  }
+  // Finds which suffix to use
+  var exp = Math.floor(Math.log(big) / Math.log(base));
+  // Divides the bumber by the equivalent suffix number
+  var val = big / Math.pow(base, exp);
+  // Keeps the number to 2 decimal places and adds the suffix
+  return val.toFixed(2) + suffixes[exp];
+}
+
+/**
+ * Converts the time to short number and adds unit
+ *
+ * @param {number} time Time in microseconds
+ * @return {string} The time with units
+ */
+function timeDuration(time) {
+  var ms, sec, min, hr, day, yr;
+  ms = sec = min = hr = day = yr = -1;
+
+  time = Math.floor(time);
+
+  // If time is 0 return a dash
+  if (time == 0) {
+    return '&mdash;';
+  }
+
+  // Obtains the milliseconds, if time is 0, return milliseconds, and units
+  ms = time % 1000;
+  time = Math.floor(time / 1000);
+  if (time == 0) {
+    return ms + 'ms';
+  }
+
+  // Obtains the seconds, if time is 0, return seconds, milliseconds, and units
+  sec = time % 60;
+  time = Math.floor(time / 60);
+  if (time == 0) {
+    return sec + 's' + '&nbsp;' + ms + 'ms';
+  }
+
+  // Obtains the minutes, if time is 0, return minutes, seconds, and units
+  min = time % 60;
+  time = Math.floor(time / 60);
+  if (time == 0) {
+    return min + 'm' + '&nbsp;' + sec + 's';
```

Comment:
Ya, there's a bit more later on (building table rows/cells, IIRC). Something we can work on later to reduce.